packages/go_router/README.md

```diff
@@ -119,6 +119,42 @@ GoRouter(
 );
 ```
 
+## Redirection
+
+You can use redirection to prevent the user from visiting a specific page. In
+go_router, redirection can be asynchronous.
+
+```dart
+GoRouter(
+  ...
+  redirect: (context, state) async {
+    if (await LoginService.of(context).isLoggedIn) {
+      return state.location;
+    }
+    return '/login';
+  },
+);
+```
+
+If the code depends on [BuildContext](https://api.flutter.dev/flutter/widgets/BuildContext-class.html)
+through the [dependOnInheritedWidgetOfExactType](https://api.flutter.dev/flutter/widgets/BuildContext/dependOnInheritedWidgetOfExactType.html)
+(which is how `of` method is usually implemented), the redirect will be called every time the [InheritedWidget](https://api.flutter.dev/flutter/widgets/InheritedWidget-class.html)
+updated.
+
+### Top-level redirect
```

Comment:
I don't think a newcomer would understand top-level vs route-level redirects by this point. Should we update the code sample above to contain both a top-level and route-level redirect?